GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m

```diff
@@ -882,17 +883,30 @@ - (void)application:(GULApplication *)application
       didReceiveRemoteNotificationWithCompletionIMP =
           [didReceiveRemoteNotificationWithCompletionIMPPointer pointerValue];
 
+  dispatch_group_t __block callbackGroup = dispatch_group_create();
+  UIBackgroundFetchResult __block latestFetchResult = UIBackgroundFetchResultNoData;
+  dispatch_group_notify(callbackGroup, dispatch_get_main_queue(), ^() {
+    completionHandler(latestFetchResult);
+  });
   // Notify interceptors.
   [GULAppDelegateSwizzler
       notifyInterceptorsWithMethodSelector:methodSelector
                                   callback:^(id<GULApplicationDelegate> interceptor) {
+                                    dispatch_group_enter(callbackGroup);
+                                    void (^localCompletionHandler)(UIBackgroundFetchResult) =
+                                        ^void(UIBackgroundFetchResult fetchResult) {
+                                          latestFetchResult = fetchResult;
```

Comment:
I think we can improve combining the fetch results. Perhaps if there is at least on `UIBackgroundFetchResultNewData` the combined result should be also `UIBackgroundFetchResultNewData`. Also I would pass `UIBackgroundFetchResultFailed` only if all calls failed.